markdown_it/extensions/tasklists/__init__.py

```diff
@@ -0,0 +1,146 @@
+"""Builds task/todo lists out of markdown lists with items starting with [ ] or [x]"""
+
+# Ported by Wolmar Nyberg Åkerström from https://github.com/revin/markdown-it-task-lists
+# ISC License
+# Copyright (c) 2016, Revin Guillen
+#
+# Permission to use, copy, modify, and/or distribute this software for any
+# purpose with or without fee is hereby granted, provided that the above
+# copyright notice and this permission notice appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+from math import ceil
+from random import random
+from typing import List
+
+from markdown_it import MarkdownIt
+
+from markdown_it.token import Token
+
+
+def tasklists_plugin(
+    md: MarkdownIt,
+    enabled: bool = False,
+    label: bool = False,
+    label_after: bool = False,
+):
```

Comment:
nearly there!

If you could add a docstring here, explaining the plugin and the parameters, so that it will show up in the documentation.

something like

```suggestion
def tasklists_plugin(
    md: MarkdownIt,
    enabled: bool: bool = False,
    label: bool = False,
    label_after: bool = False,
):
   """A plugin which parses GitHub style checkbox TODO lists.
   
    For example::
   
       - [ ] An item that needs doing
       - [x] An item that is complete
   
   :param enabled: this does...
   :param label: this does...
   :param label_after: this does...

   """
```

Reply:
oh you did this just as I was typing this lol.
Anyhow, perhaps adding the actual example above would be good

Reply:
Good idea, I added the example to the docstring.